File: temp/ComputerStore/library/src/General/Address.cpp
#include "General/Address.h"

Address::Address(std::string street, std::string number) : street(street), number(number) {}

Address::~Address() {

}
std::string Address::toString() {
    return "Address: street = "+this->street+", house number = "+this->number;
}

const std::string &Address::getStreet() const {
    return street;
}

const std::string &Address::getNumber() const {
    return number;
}

File: src/main/java/model/entities/Address.java
package model.entities;

import controller.exceptions.user.AddressException;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@ToString
public class Address {
    
    private String street;

    private String number;

    public Address() {
    }

    public Address(String street, String number) throws AddressException {

        if (street == null || number == null)
            throw new AddressException(AddressException.NULL_FIELD);

        if (street.equals("") || number.equals(""))
            throw new AddressException(AddressException.EMPTY_FIELD);

        this.street = street;
        this.number = number;
    }
}
